src/sidebar/services/test/groups-test.js

```diff
@@ -253,6 +253,93 @@ describe('groups', function() {
         });
       });
     });
+
+    it('shows service groups when logged in', () => {
+      fakeSettings.services = [{ groups: ['abc123'] }];
+      const svc = service();
+
+      // Create groups response from server.
+      const groups = [{ name: 'Public', id: '__world__' }, { name: 'ABC', id: 'abc123'}];
+
+      fakeApi.groups.list.returns(Promise.resolve({
+        token: '1234',
+        data: groups,
+      }));
+
+      return svc.load().then(groups => {
+        const onlyGroupABCShown = groups.every(g => g.id === 'abc123');
+        assert.equal(onlyGroupABCShown, true);
```

Comment:
I would recommend using `assert.deepEqual(actualArray, expectedArray)` to compare arrays. You'll get more information about what went wrong if the test fails that way. If order is not important, you can sort the arrays beforehand.